lib/sda/host_test.go

```diff
@@ -1,15 +1,76 @@
 package sda_test
 
 import (
+	"golang.org/x/net/context"
 	"testing"
 	"time"
 
+	"github.com/dedis/cothority/lib/cosi"
 	"github.com/dedis/cothority/lib/dbg"
 	"github.com/dedis/cothority/lib/network"
 	"github.com/dedis/cothority/lib/sda"
+	_ "github.com/dedis/cothority/protocols"
+	"github.com/dedis/crypto/config"
 	"github.com/satori/go.uuid"
 )
 
+// Test hacky way of launching CoSi protocol externally
+func TestHackyCosi(t *testing.T) {
+	// TODO: github.com/btcsuite leaks 2 go routine and is imported by byzcoin
+	// in blkparser.
+	//	defer dbg.AfterTest(t)
+	dbg.TestOutput(testing.Verbose(), 4)
+	// make up the hosts and the entity list
+	h1 := sda.NewLocalHost(2000)
+	h1.ListenNoblock()
+	h1.StartProcessMessages()
+	defer h1.Close()
+	h2 := sda.NewLocalHost(2010)
+	h2.Listen()
+	h2.StartProcessMessages()
+	defer h2.Close()
+	h3 := sda.NewLocalHost(2020)
+	h3.Listen()
+	h3.StartProcessMessages()
+	defer h3.Close()
```

Comment:
It's not clear for me what all theses boolean flags are when you call "GenLocalTree(...)", I was lazy to dive into the "local api" to get to know exactly what's happenning.